h2o-parsers/h2o-parquet-parser/src/main/java/water/parser/parquet/ChunkConverter.java

```diff
@@ -84,22 +87,34 @@ private PrimitiveConverter newConverter(int colIdx, byte vecType, PrimitiveType
   }
 
   private static class StringConverter extends PrimitiveConverter {
-
+    // Maximum array size (Integer.MAX_VALUE - 8) minus one place for a trailing zero
     private final BufferedString _bs = new BufferedString();
     private final int _colIdx;
     private final WriterDelegate _writer;
     private final boolean _dictionarySupport;
     private String[] _dict;
+    private long _writtenAmount;
+    private boolean _overflew;
 
     StringConverter(WriterDelegate writer, int colIdx, boolean dictionarySupport) {
       _colIdx = colIdx;
       _writer = writer;
       _dictionarySupport = dictionarySupport;
+      _writtenAmount = 0;
+      _overflew = false;
     }
 
     @Override
     public void addBinary(Binary value) {
+      if(_overflew) return;
```

Comment:
The case when the string comes from a dictionary is not handled, see method:

    public void addValueFromDictionary(int dictionaryId)